### PR TITLE
nextflow: init at 21.10.6

### PR DESCRIPTION
**Pull request by Etjean**
###### Motivation for this change
Addition of [nextflow](https://github.com/nextflow-io/nextflow).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - => Tested with `nextflow run hello` in a pure nix-shell.
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/nextflow/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  pname = "nextflow";
+  version = "21.10.6";
+
+  src = fetchurl {
+    url = "https://github.com/nextflow-io/nextflow/releases/download/v${version}/nextflow-${version}-all";
+    sha256 = "0l9hi51vrhvfx3px2pxw7lp4h21n8ks50x4icfk3hbgl2hwf7fvx";
+  };
+
+  nativeBuildInputs = [makeWrapper];
+  buildInputs = [jre wget which];
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:20]: For consistency with the rest of nixpkgs:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
  buildInputs = [ jre wget which ];
```
btw, you can automatically do trivial formatting stuff with `nixpkgs-fmt`.

File: pkgs/development/interpreters/nextflow/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  dontUnpack = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    chmod +x $src
+    mv $src $out/bin/$pname
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:27]:
```suggestion
    install -Dm755 $src $out/bin/nextflow
```
Usage of `pname` here makes it a bit harder to read IMO, with no benefits.

File: pkgs/development/interpreters/nextflow/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  '';
+
+  postFixup = ''
+  wrapProgram $out/bin/${pname} --prefix PATH : ${lib.makeBinPath buildInputs}
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:31]: Indentation seems off:
```suggestion
    wrapProgram $out/bin/nextflow --prefix PATH : ${lib.makeBinPath buildInputs}
```

File: pkgs/development/interpreters/nextflow/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  '';
+
+  meta = with lib; {
+    description = "A DSL for data-driven computational pipelines";
+    longDescription = "Nextflow is a bioinformatics workflow manager that enables the development of portable and reproducible workflows. It supports deploying workflows on a variety of execution platforms including local, HPC schedulers, AWS Batch, Google Cloud Life Sciences, and Kubernetes. Additionally, it provides support for manage your workflow dependencies through built-in support for Conda, Docker, Singularity, and Modules.";
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:36]:
```suggestion
    longDescription = ''
      Nextflow is a bioinformatics workflow manager that enables the development of portable and reproducible workflows. 
      It supports deploying workflows on a variety of execution platforms including local, HPC schedulers, AWS Batch, Google Cloud Life Sciences, and Kubernetes. 
      Additionally, it provides support for manage your workflow dependencies through built-in support for Conda, Docker, Singularity, and Modules. 
    '';
```

File: pkgs/development/interpreters/nextflow/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    homepage = "https://www.nextflow.io/";
+    changelog = "https://github.com/nextflow-io/nextflow/releases";
+    license = licenses.asl20;
+    maintainers = [maintainers.Etjean];
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:40]:
```suggestion
    maintainers = [ maintainers.Etjean ];
```

File: pkgs/development/interpreters/nextflow/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    mkdir -p $out/bin
+    chmod +x $src
+    mv $src $out/bin/$pname
+  '';
+
+  postFixup = ''
+  wrapProgram $out/bin/${pname} --prefix PATH : ${lib.makeBinPath buildInputs}
+  '';
+
+  meta = with lib; {
```
Comment (IvarWithoutBones) [inline pkgs/development/interpreters/nextflow/default.nix:34]: Could you add `meta.mainProgram`?

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]

**Comment by Etjean** [comment]
Thanks for the review @IvarWithoutBones, I incorporated all suggestions.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Just a few more suggestions :)

**Review by IvarWithoutBones — APPROVED** [review]
LGTM, thanks for the work :+1: 

Result of `nixpkgs-review pr 152961` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nextflow</li>
  </ul>
</details>

Binary also shows usage

**Pull request merged**